app/lib/registries/full_topic_taxonomy_registry.rb

```diff
@@ -1,22 +1,34 @@
 module Registries
-  class FullTopicTaxonomyRegistry
-    CACHE_KEY = "#{NAMESPACE}/full_topic_taxonomy".freeze
+  class FullTopicTaxonomyRegistry < Registry
+    include CacheableRegistry
 
     def [](base_path)
       taxonomy[base_path]
     end
 
     def taxonomy
-      @taxonomy ||= Rails.cache.fetch(CACHE_KEY, expires_in: 1.hour) do
-        taxonomy_hash
+      @taxonomy ||= Rails.cache.fetch(cache_key) do
+        cacheable_data
       end
```

Comment:
As this logic is the same for all the cacheable registries, I wonder if there should be a `fetch_from_cache` method in `CacheableRegistry` which does this.

Then this method would be

```ruby
def taxonomy
  @taxonomy = fetch_fron_cache
```

Reply:
Thanks, that's a good idea. I've added a commit to do this.